refactor: Modernize NULL to nullptr

bobtista opened the pull request:
- Used clang-tidy to consistently change NULL->nullptr
- Added include for cppMacros to files that use nullptr and didn't have it (required for VC6)

Tested build and played a game successfully.

xezon commented on Core/Libraries/Source/WWVegas/WW3D2/decalsys.cpp:
```diff
@@ -47,6 +47,7 @@
  *   DecalGeneratorClass::Set_Mesh_Transform -- sets the current mesh coordinate system        *
  * - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - */
 
+#include <Utility/CppMacros.h>
```
These includes should not be necessary in most cases. The include should come through WWCommon already. If not, then an always.h (WWVegas) or PreRTS.h (GameEngine) include is missing somewhere. GameEngineDevice does not use PreRTS.h there is may be necessary sometimes, but I would expect not in many cases because WWVegas stuff gets always.h

xezon reviewed (commented):
Adding the CppMacros include everwhere should not be necessary.

xezon reviewed (commented):
When I search for NULL across whole project, then it finds 20231 NULLs, but this change only touched 8724 NULLs. Why is there this large discrepancy?

xezon commented:
But why is it 35000 lines? When I search in VS Code I get 20000 hits for NULL. Are there 15000 zeros for null?

bobtista commented:
Ok I think we got it done :) All NULLs (other than ones inside comments or strings) are modernized. I comment out NULL definitions, and builds are passing.

On main, there are 37,524 occurrences of "NULL" (grep -rw "NULL" . | wc -l) - only about 100 of them were converted to 0 or False, the rest are now nullptr.

All 1,750 remaining NULLs are intentionally left.

- 1,275 in comments/documentation
- 250 in debug/assert messages (error strings)
- 38 in string literals (identifiers/logging)
- 187 in names/constants (Windows API, enums, function names)

xezon commented:
Holy shit. This change is massive... I reached WWLib, maybe 10%... did you review this refactor?

Notes so far:

#include <Utility/CppMacros.h> in GameMemoryNull.h, FEBDispatch.h, LANPlayer.h, simpleplayer.cpp, urllaunch.cpp, ... are necessary?
In the debug and profile lib the include can likely be put into some shared header, not added to every single cpp.
The amount of CppMacros.h includes needs to be minimized. There are many unnecessary ones.

There are NULLs in simpleplayer.cpp, MiniDumper.cpp, maybe more

NULLs in commented code in Debug.cpp and GameMemory.cpp, XferSave.cpp, ConnectionManager.cpp, FirewallHelper.cpp, LobbyUtils.cpp, ... and others.

Empty lines added in BaseHeightMap.cpp, FlatHeightMap.cpp, W3DPropBuffer.cpp, W3DTerrainBackground.cpp, W3DTreeBuffer.cpp, W3DWater.cpp

```
TileData *WorldHeightMap::m_alphaTiles[NUM_ALPHA_TILES]={nullptr,nullptr,nullptr,nullptr,nullptr,nullptr,nullptr,nullptr,nullptr,nullptr,nullptr,nullptr};
```
can simplify to
```
TileData *WorldHeightMap::m_alphaTiles[NUM_ALPHA_TILES]={0};
```

NULL removed:
```
-    \param frameName name of frame to use (NULL for global)
+    \param frameName name of frame to use ( for global)
```

This is bug in Constructor: How did this happen?
```
-	CollidedRenderObj(NULL)
+	CollidedRenderObj()
```

This casting should not be necessary
```
DeviceName(static_cast<const TCHAR*>(nullptr)),
DeviceVendor(static_cast<const TCHAR*>(nullptr)),
DevicePlatform(static_cast<const TCHAR*>(nullptr)),
DriverName(static_cast<const TCHAR*>(nullptr)),
DriverVendor(static_cast<const TCHAR*>(nullptr)),
DriverVersion(static_cast<const TCHAR*>(nullptr)),
HardwareName(static_cast<const TCHAR*>(nullptr)),
HardwareVendor(static_cast<const TCHAR*>(nullptr)),
HardwareChipset(static_cast<const TCHAR*>(nullptr))
```

This needs another review pass.

Why does this change add around 700 lines, according to GitHub?

xezon commented:
Are all these includes really necessary?

```
Core\GameEngine\Include\Common\Debug.h(48):#include <Utility/CppMacros.h>
Core\GameEngine\Include\GameNetwork\LANPlayer.h(30):#include <Utility/CppMacros.h>
Core\GameEngine\Include\GameNetwork\WOLBrowser\FEBDispatch.h(31):#include <Utility/CppMacros.h>
Core\GameEngine\Source\Common\System\MiniDumper.cpp(19):#include <Utility/CppMacros.h>
Core\GameEngine\Source\Common\WorkerProcess.cpp(19):#include <Utility/CppMacros.h>
Core\Libraries\Source\Compression\EAC\huffencode.cpp(24):#include <Utility/CppMacros.h>
Core\Libraries\Source\debug\debug_except.cpp(29):#include <Utility/CppMacros.h>
Core\Libraries\Source\debug\debug_internal.cpp(30):#include <Utility/CppMacros.h>
Core\Libraries\Source\debug\debug_macro.h(31):#include <Utility/CppMacros.h>
Core\Libraries\Source\debug\debug_stack.cpp(30):#include <Utility/CppMacros.h>
Core\Libraries\Source\debug\internal.h(32):#include <Utility/CppMacros.h>
Core\Libraries\Source\debug\internal_io.h(31):#include <Utility/CppMacros.h>
Core\Libraries\Source\profile\internal_funclevel.h(31):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WW3D2\aabtree.cpp(65):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WW3D2\aabtree.h(41):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WW3D2\agg_def.cpp(36):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WW3D2\animatedsoundmgr.cpp(40):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WW3D2\animatedsoundmgr.h(41):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WW3D2\collect.cpp(74):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WW3D2\coltest.h(42):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WW3D2\decalsys.cpp(50):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WW3D2\decalsys.h(40):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WW3D2\distlod.cpp(80):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WW3D2\distlod.h(38):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WW3D2\dx8texman.h(41):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WW3D2\dx8webbrowser.cpp(33):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WW3D2\dynamesh.h(36):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WW3D2\hanim.cpp(39):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WW3D2\hanim.h(39):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WW3D2\hcanim.cpp(52):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WW3D2\htree.cpp(54):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WW3D2\matinfo.cpp(38):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WW3D2\matinfo.h(38):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WW3D2\meshdam.cpp(41):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WW3D2\pivot.cpp(41):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WW3D2\pivot.h(39):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WW3D2\pointgr.h(38):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WW3D2\proto.h(38):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WW3D2\rddesc.h(38):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WW3D2\render2dsentence.cpp(37):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WW3D2\renderobjectrecycler.cpp(44):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WW3D2\rendobj.cpp(75):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WW3D2\rendobj.h(42):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WW3D2\ringobj.cpp(74):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WW3D2\shattersystem.cpp(39):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WW3D2\soundrobj.h(38):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WW3D2\sphereobj.cpp(71):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WW3D2\streak.h(38):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WW3D2\texproject.cpp(77):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WW3D2\texproject.h(41):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WW3D2\texture.cpp(42):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WW3D2\texture.h(43):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WW3D2\textureloader.cpp(41):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WW3D2\textureloader.h(41):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WW3D2\texturethumbnail.cpp(19):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WW3D2\visrasterizer.cpp(39):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWAudio\AABTreeSoundCullClass.h(38):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWAudio\AudioEvents.h(38):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWAudio\PriorityVector.h(34):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWAudio\sound3dhandle.cpp(37):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWAudio\SoundBuffer.cpp(38):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWAudio\SoundCullObj.h(38):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWAudio\SoundScene.cpp(38):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWAudio\SoundScene.h(38):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWAudio\SoundSceneObj.cpp(36):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWAudio\SoundSceneObj.h(36):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWAudio\Threads.cpp(33):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWAudio\Threads.h(34):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWAudio\Utils.h(38):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWAudio\WWAudio.cpp(38):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWAudio\WWAudio.h(38):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWDebug\wwdebug.cpp(45):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWDebug\wwmemlog.cpp(41):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWDebug\wwprofile.cpp(52):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWDownload\FTP.CPP(26):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWDownload\registry.cpp(23):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWLib\argv.h(39):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWLib\binheap.h(38):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWLib\BSEARCH.H(20):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWLib\chunkio.cpp(65):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWLib\cpudetect.cpp(19):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWLib\DbgHelpLoader.cpp(19):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWLib\Except.cpp(52):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWLib\Except.h(38):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWLib\FastAllocator.h(31):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWLib\hash.cpp(38):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWLib\hash.h(39):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWLib\HASHLIST.H(46):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWLib\INDEX.H(53):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWLib\INI.H(38):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWLib\inisup.h(38):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWLib\mempool.h(46):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWLib\multilist.h(40):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWLib\ntree.h(38):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWLib\RAMFILE.H(38):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWLib\RAWFILE.H(42):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWLib\rcfile.h(38):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWLib\refcount.cpp(42):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWLib\registry.cpp(36):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWLib\registry.h(37):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWLib\sharebuf.h(38):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWLib\Signaler.h(41):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWLib\simplevec.h(51):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWLib\SLIST.H(55):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWLib\STLUtils.h(25):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWLib\strtok_r.cpp(40):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWLib\systimer.h(38):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWLib\tgatodxt.cpp(40):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWLib\trect.h(38):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWLib\uarray.h(41):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWLib\Vector.H(55):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWLib\WWCOMUtil.cpp(38):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWLib\wwstring.h(38):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWLib\XPIPE.H(38):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWLib\XSTRAW.H(38):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWMath\aabox.h(56):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWMath\aabtreecull.cpp(38):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWMath\aabtreecull.h(38):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWMath\cullsys.cpp(37):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWMath\cullsys.h(38):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWMath\curve.h(38):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWMath\gridcull.cpp(57):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWMath\gridcull.h(55):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWMath\lookuptable.cpp(39):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWMath\lookuptable.h(40):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWMath\matrix3d.cpp(58):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWMath\obbox.h(45):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWMath\ode.cpp(42):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWMath\quat.cpp(52):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWMath\quat.h(39):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWMath\vector2.h(53):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWMath\vector3.h(67):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWSaveLoad\definitionmgr.cpp(37):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWSaveLoad\definitionmgr.h(38):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWSaveLoad\parameter.h(40):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWSaveLoad\parameterlist.h(38):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWSaveLoad\saveload.cpp(38):#include <Utility/CppMacros.h>
Core\Libraries\Source\WWVegas\WWSaveLoad\saveload.h(38):#include <Utility/CppMacros.h>
GeneralsMD\Code\GameEngineDevice\Include\W3DDevice\GameClient\W3DShadow.h(26):#include <Utility/CppMacros.h>
GeneralsMD\Code\GameEngineDevice\Include\W3DDevice\GameClient\W3DShroud.h(26):#include <Utility/CppMacros.h>
GeneralsMD\Code\GameEngine\Include\GameClient\GameText.h(44):#include <Utility/CppMacros.h>
GeneralsMD\Code\GameEngine\Include\GameClient\Line2D.h(31):#include <Utility/CppMacros.h>
GeneralsMD\Code\GameEngine\Include\GameLogic\ScriptActions.h(31):#include <Utility/CppMacros.h>
GeneralsMD\Code\GameEngine\Source\Common\System\encrypt.cpp(29):#include <Utility/CppMacros.h>
GeneralsMD\Code\Libraries\Source\WWVegas\WW3D2\aabtreebuilder.cpp(60):#include <Utility/CppMacros.h>
GeneralsMD\Code\Libraries\Source\WWVegas\WW3D2\aabtreebuilder.h(40):#include <Utility/CppMacros.h>
GeneralsMD\Code\Libraries\Source\WWVegas\WW3D2\animobj.h(42):#include <Utility/CppMacros.h>
GeneralsMD\Code\Libraries\Source\WWVegas\WW3D2\assetmgr.cpp(80):#include <Utility/CppMacros.h>
GeneralsMD\Code\Libraries\Source\WWVegas\WW3D2\assetmgr.h(39):#include <Utility/CppMacros.h>
GeneralsMD\Code\Libraries\Source\WWVegas\WW3D2\boxrobj.cpp(91):#include <Utility/CppMacros.h>
GeneralsMD\Code\Libraries\Source\WWVegas\WW3D2\camera.cpp(74):#include <Utility/CppMacros.h>
GeneralsMD\Code\Libraries\Source\WWVegas\WW3D2\camera.h(47):#include <Utility/CppMacros.h>
GeneralsMD\Code\Libraries\Source\WWVegas\WW3D2\dazzle.cpp(41):#include <Utility/CppMacros.h>
GeneralsMD\Code\Libraries\Source\WWVegas\WW3D2\decalmsh.cpp(54):#include <Utility/CppMacros.h>
GeneralsMD\Code\Libraries\Source\WWVegas\WW3D2\dx8indexbuffer.cpp(41):#include <Utility/CppMacros.h>
GeneralsMD\Code\Libraries\Source\WWVegas\WW3D2\dx8renderer.cpp(43):#include <Utility/CppMacros.h>
GeneralsMD\Code\Libraries\Source\WWVegas\WW3D2\dx8renderer.h(41):#include <Utility/CppMacros.h>
GeneralsMD\Code\Libraries\Source\WWVegas\WW3D2\dx8vertexbuffer.cpp(42):#include <Utility/CppMacros.h>
GeneralsMD\Code\Libraries\Source\WWVegas\WW3D2\dx8wrapper.cpp(53):#include <Utility/CppMacros.h>
GeneralsMD\Code\Libraries\Source\WWVegas\WW3D2\dx8wrapper.h(43):#include <Utility/CppMacros.h>
GeneralsMD\Code\Libraries\Source\WWVegas\WW3D2\hanimmgr.cpp(49):#include <Utility/CppMacros.h>
GeneralsMD\Code\Libraries\Source\WWVegas\WW3D2\hlod.cpp(129):#include <Utility/CppMacros.h>
GeneralsMD\Code\Libraries\Source\WWVegas\WW3D2\hmorphanim.cpp(39):#include <Utility/CppMacros.h>
GeneralsMD\Code\Libraries\Source\WWVegas\WW3D2\hrawanim.cpp(53):#include <Utility/CppMacros.h>
GeneralsMD\Code\Libraries\Source\WWVegas\WW3D2\linegrp.h(40):#include <Utility/CppMacros.h>
GeneralsMD\Code\Libraries\Source\WWVegas\WW3D2\mesh.cpp(89):#include <Utility/CppMacros.h>
GeneralsMD\Code\Libraries\Source\WWVegas\WW3D2\mesh.h(39):#include <Utility/CppMacros.h>
GeneralsMD\Code\Libraries\Source\WWVegas\WW3D2\meshbuild.cpp(61):#include <Utility/CppMacros.h>
GeneralsMD\Code\Libraries\Source\WWVegas\WW3D2\meshgeometry.cpp(88):#include <Utility/CppMacros.h>
GeneralsMD\Code\Libraries\Source\WWVegas\WW3D2\meshgeometry.h(40):#include <Utility/CppMacros.h>
GeneralsMD\Code\Libraries\Source\WWVegas\WW3D2\meshmatdesc.cpp(39):#include <Utility/CppMacros.h>
GeneralsMD\Code\Libraries\Source\WWVegas\WW3D2\meshmatdesc.h(40):#include <Utility/CppMacros.h>
GeneralsMD\Code\Libraries\Source\WWVegas\WW3D2\meshmdlio.cpp(78):#include <Utility/CppMacros.h>
GeneralsMD\Code\Libraries\Source\WWVegas\WW3D2\part_buf.cpp(37):#include <Utility/CppMacros.h>
GeneralsMD\Code\Libraries\Source\WWVegas\WW3D2\part_emt.cpp(40):#include <Utility/CppMacros.h>
GeneralsMD\Code\Libraries\Source\WWVegas\WW3D2\part_emt.h(38):#include <Utility/CppMacros.h>
GeneralsMD\Code\Libraries\Source\WWVegas\WW3D2\render2d.h(38):#include <Utility/CppMacros.h>
GeneralsMD\Code\Libraries\Source\WWVegas\WW3D2\scene.cpp(64):#include <Utility/CppMacros.h>
GeneralsMD\Code\Libraries\Source\WWVegas\WW3D2\sortingrenderer.cpp(41):#include <Utility/CppMacros.h>
GeneralsMD\Code\Libraries\Source\WWVegas\WW3D2\vertmaterial.cpp(40):#include <Utility/CppMacros.h>
GeneralsMD\Code\Libraries\Source\WWVegas\WW3D2\ww3d.cpp(84):#include <Utility/CppMacros.h>
GeneralsMD\Code\Libraries\Source\WWVegas\WW3D2\ww3d.h(38):#include <Utility/CppMacros.h>
```

bobtista commented:
> Are all these includes really necessary?
> 
> ```
> Core\GameEngine\Include\Common\Debug.h(48):#include <Utility/CppMacros.h>
> ...
> ```

I made a script that checked each of the files that we added cppmacros includes to, recursively going through their includes to see if they indirectly have it, caching the results, then removed it from files where it was redundant - it was about half. Then I tried seeing if any files were commonly included in the remaining ones, like maybe we can add cppmacros to another commonly imported one. I didn't find anything great though

xezon commented:
This change still needs work. If you cannot reduce amount of includes then I can take care of it.

xezon commented:
65 is still quite a big number. I wonder if we need a new strategy to inject this header into all code.

bobtista commented:
HeightMap.cpp has indent changes: fixed

> WOW64_FLOATING_SAVE_AREA was changed to FLOATING_SAVE_AREA. Why?

It's something clang-tidy does when I enable --fix-errors I think. I assumed it had to do with me running it on a windows x64 machine. Reverted it. 

I found a lot of places where comments were changed to nullptr and made a push to have the comments say "null" where it makes more sense. 

define NULL commented: cleaned these up

NULL_HANDLE: reverted those usages but kept the updated macro definition where it was NULL → nullptr.

Indentation changes (eg autorun.cpp) restored.

Character chaos in Jsupport.cpp restored to main and fixed

Missing initialization eg in WSYS_RAMFile.cpp fixed

core_utility stuff cleaned up

Can do {0}: done (this was in comments)

m_nextOpen is ptr: fixed

File edits with no other changes - removed whitespace changes so only the null changes from that commit are here.

RE: Removed code - This was probably a merge conflict or something before, it's no longer on this branch or in main
@@ -642 +641,0 @@ int CWorldBuilderApp::ExitInstance()
-	TextureLoadTaskClass::shutdown();

bobtista commented:
Did you fetch latest?
